### ISSUE TITLE
Helping out with automation

### ISSUE THREAD

**Issue by Walter-o** [DevilXD/TwitchDropsMiner#257]
Hey!
I have been using this program for some time now and added a bunch of external features to it over time.

TLDR: I solved some problems like auto-restart, multi-accounts, docker, etc but idk where to post the information or files that can help people.

### List of things from main readme that i made work:

- _It being possible to run it without a GUI, or with only a console attached._
I still run it as GUI so i can control it, but inside a virtual display with xvfb, and it's accessible via any VNC viewer program
> ![image](https://github.com/DevilXD/TwitchDropsMiner/assets/33218378/3f64c0bf-9f33-417a-a90c-bb2c00469b0a)

- _Any form of automatic restart when an error happens._
I edited main.py to auto-exit whenever an error happens and set docker's restart policy to automatically restart when it fails
The only change in main.py: 
> ![image](https://github.com/DevilXD/TwitchDropsMiner/assets/33218378/c948476b-2016-42d3-97ab-d29c22e2826b)

- _Docker or any other form of remote deployment._
yes i managed to cram it into docker with only modifying constants.py slightly (to change file paths for cookies, cache etc)
(i also had to pin the version of aiohttp in requirements.txt since a newer version changed cookie files format)
> ![image](https://github.com/DevilXD/TwitchDropsMiner/assets/33218378/404110ae-6856-414e-9657-47987fb5d48a)

- _Using it with more than one managed account._
With docker, adding an account is just adding a few lines in docker-compose.yml and logging the account in.
The only downside is that every new user adds around 200 MB ram, but you do get the separation.
> ![image](https://github.com/DevilXD/TwitchDropsMiner/assets/33218378/bc1bd821-3436-4182-b771-08c80d1c5aaa)
> ![image](https://github.com/DevilXD/TwitchDropsMiner/assets/33218378/bfaac367-edac-4130-9172-a79ca8653491)
> ![image](https://github.com/DevilXD/TwitchDropsMiner/assets/33218378/0a082e55-affe-49d8-b0d9-5dd1285e6c33)

- _Being hosted on a remote server as a 24/7 miner._
yes, i'm currently using a RPI4B 8GB since it only consumes 3-6 watts of electricity

### Things i didn't get working yet or haven't tried:
- _Unattended operation_
It can be set to auto-update from this github repo via watchtower but that's only safe when using something like kata-containers.

# Conclusion
#### I don't really know what could replace it but yeah i just figured i'd post this at the very least. 👍

**Comment by Walter-o** [comment]
Very well then.
